src/werkzeug/sansio/multipart.py

```diff
@@ -0,0 +1,251 @@
+import re
+from dataclasses import dataclass
+from enum import auto
+from enum import Enum
+from typing import cast
+from typing import List
+from typing import Optional
+from typing import Tuple
+from typing import Union
+
+from .._internal import _to_bytes
+from .._internal import _to_str
+from ..datastructures import Headers
+from ..exceptions import RequestEntityTooLarge
+from ..http import parse_options_header
+
+
+@dataclass
+class Preamble:
+    data: bytes
+
+
+@dataclass
+class Field:
+    name: str
+    headers: Headers
+
+
+@dataclass
+class File:
+    name: str
+    filename: str
+    headers: Headers
+
+
+@dataclass
+class Data:
+    data: bytes
+    more_data: bool
+
+
+@dataclass
+class Epilogue:
+    data: bytes
+
+
+@dataclass
```

Comment:
Don't need to decorate this one. Could also treat it as a singleton?

Reply:
Decorating gives a nicer repr though. Happy to, but I felt a consistent `isinstance(event, Type[Event])` check would be clearer. Happy to change.

Reply:
Changed